crates/subspace-gateway/src/commands/run.rs

```diff
@@ -102,7 +102,7 @@ pub async fn run(run_options: RunOptions) -> anyhow::Result<()> {
         Semaphore::new(out_connections as usize * PIECE_PROVIDER_MULTIPLIER),
     );
     let piece_getter = DsnPieceGetter::new(piece_provider);
-    let object_fetcher = ObjectFetcher::new(piece_getter, erasure_coding, Some(max_size));
+    let object_fetcher = ObjectFetcher::new(piece_getter.into(), erasure_coding, Some(max_size));
```

Comment:
This puzzled me for a moment, this is the first time I saw `From` used instead of `Arc::new()` :thinking: